src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt

```diff
@@ -51,9 +63,41 @@ class KediatrHandlerClassMarker : LineMarkerProvider {
     }
 
     private fun navigationHandler(superClassQualifiedNames: Collection<String>, commandTypeName: String) =
-        GutterIconNavigationHandler<PsiElement> { _, elt ->
+        GutterIconNavigationHandler<PsiElement> { me, elt ->
             val project = elt.project
             val handler = project.service<HandlerService>().findHandler(elt, superClassQualifiedNames, commandTypeName)
-            handler.firstOrNull()?.navigate(false)
+            handler.firstOrNull()?.navigate(false) ?: showPopupWithIcon(elt, me)
         }
+
+    private fun showPopup(element: PsiElement, me: MouseEvent) {
+        val mainClass = (element.parent as KtFile).classes.firstOrNull() ?: return
+        val title = POPUP_TITLE_PREFIX + mainClass.name
+        val popup = JBPopupFactory.getInstance()
+            .createPopupChooserBuilder<Any>(listOf(" + $POPUP_STEP_MESSAGE"))
+            .setTitle(title)
+            .setMovable(false)
+            .setResizable(false)
+            .setRequestFocus(true)
+            .setCancelOnWindowDeactivation(false)
+            .setItemChosenCallback { element.project.service<CreateNewHandlerService>().create(element) }
+            .createPopup()
+        popup.show(RelativePoint(Point(me.xOnScreen + 30, me.yOnScreen + 20)))
+    }
+
+    private fun showPopupWithIcon(element: PsiElement, me: MouseEvent) {
+        val mainClass = (element.parent as KtFile).classes.firstOrNull() ?: return
```

Comment:
This doesn't work if file contains more than one classes ie:

```
class ExistingCommand: Command

class ClassIWantToCreateHandlerFor : Command
```
it tries to create handler for the existing command. Instead we could use KtClass which is already filtered in getLineMarkerInfo. Also I'm not sure this casting is safe `(element.parent as KtFile)`  🤔

Reply:
done.